File: familia.py
padres = {
    "zeus": "cronos",  
    "kratos": "zeus", 
    "atreus": "kratos",
     
    "poseidon": "cronos",
    }

madres = {
    "zeus": "rhea", 
    "kratos": "callisto", 
    "calliope": "lysandra", 
    "atreus": "laufey",
    }

def hermanos():
    hermanos = {}
    for hijo in padres:
        padre = padres[hijo]
        madre = madres.get(hijo)
        if padre and madre:

            hermanos[hijo] = [h for h in padres if padres[h] == padre or madres.get(h) == madre and h != hijo]
            for h in hermanos[hijo]:
                if h == hijo:
                    hermanos[hijo].remove(h)
            print(f"Hermanos de {hijo}: {hermanos[hijo]}")
    return hermanos

# abuelos: cronos y zeus
# orden: cronos-zeus-kratos-atreus (por ejemplo)
# primero deberia obtener los hijos e ir escalando para conseguir el padre del padre

def abuelos():
    abuelos = {}
    
    for hijo in padres:
        abuelo = padres.get("zeus")
        padre = padres[hijo]
        nieto = hijo

        abuelos[nieto] = [abuelo]
        print(abuelos)
    return abuelos

            
abuelos()

File: familia.pl
es_padre(cronos,zeus).
es_padre(cronos,poseidon).
es_padre(zeus,kratos).
es_padre(kratos,calliope).
es_padre(kratos,atreus).

es_madre(rhea,zeus).
es_madre(callisto,kratos).
es_madre(freyja,baldur).
es_madre(lysandra,calliope).
es_madre(laufey,atreus).

es_tio(poseidon,kratos) :- es_padre(zeus,kratos),es_hermano(poseidon,zeus).

es_hermano(calliope,atreus) :- es_padre(kratos,calliope),es_padre(kratos,atreus).
es_hermano(poseidon,zeus) :- es_padre(cronos,zeus),es_padre(cronos,poseidon).


es_abuelo(cronos,kratos) :- es_padre(cronos,zeus),es_padre(zeus,kratos).
es_abuelo(zeus,calliope) :- es_padre(kratos,calliope),es_padre(zeus,kratos).
es_abuelo(zeus,atreus) :- es_padre(kratos,atreus),es_padre(zeus,kratos).


es_abuela(rhea,kratos) :- es_madre(rhea,zeus),es_padre(zeus,kratos).
es_abuela(callisto,calliope) :- es_padre(kratos,calliope),es_madre(callisto,kratos).
es_abuela(callisto,atreus) :- es_padre(kratos,atreus),es_madre(callisto,kratos).

% CONSULTAS

% es_padre(cronos,zeus). true
% es_madre(laufey,atreus). true
% es_tio(poseidon,kratos). true
% es_hermano(poseidon,zeus). true
% es_abuelo(zeus,calliope). true
% es_abuela(rhea,kratos). true